### PR TITLE
Docs update function develop

### PR DESCRIPTION
**Pull request by choir27**
## What does this PR do?

- Separate `bodyText` and `bodyJson` in codesnippets
- We should also remove mention of bodyRaw
- Add binary examples that return image from function
- Logging section
   - add how in most of the languages, we support spread operator
   - add relevant code samples ie `log("Name", nickname, "Age", age)/error("error", error, "message", message )`
- Add variables to accessing environment variables table
   - APPWRITE_FUNCTION_API_ENDPOINT 
   - APPWRITE_VERSION APPWRITE_REGION
   - APPWRITE_FUNCTION_API_KEY 
- Add Go under dependencies
- Fix [] wrapped placeholder variables
- Using with JWT
   - update copy for when there is no JWT token available
- Add Go to code structure
- Add Go code samples

## Test Plan

Manually test code snippets and code structure

## Related PRs and Issues

X

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 1]
```diff
@@ -1504,6 +1682,42 @@ export default function ({req, res, error}: any){
     return res.text("Document created");
 }
 ```
+```go
+package handler
+
+import (
+	"os"
```
Comment (loks0n) [inline src/routes/docs/products/functions/develop/+page.markdoc:1689]: Weird formatting here

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 2]
```diff
@@ -1911,7 +2166,7 @@ namespace DotNetRuntime
             if (Context.Req.Headers.ContainsKey("x-appwrite-user-jwt")) {
                 client.SetJWT(Context.Req.Headers["x-appwrite-user-jwt"]);
             } else {
-                return Context.Res.Text("Please sign in, JWT not found");
+                return Context.Res.Text("Access denied: This function requires authentication. Please sign in to continue.d");
```
Comment (loks0n) [inline src/routes/docs/products/functions/develop/+page.markdoc:2169]: continue.d looks like typo

File: src/routes/docs/products/functions/develop/+page.markdoc [hunk 3]
```diff
@@ -206,6 +206,54 @@ export default ({ req, res, log, error }: any) => {
   });
 };
 ```
+```go
+package handler
+
+import (
+	"fmt"
+	"os"
+
```
Comment (loks0n) [inline src/routes/docs/products/functions/develop/+page.markdoc:215]: Is this space here standard formatting for go?

Comment (choir27) [reply src/routes/docs/products/functions/develop/+page.markdoc:215]: Yes, this formatting is standard for go. Please take a look at the @byawitz's [playground-go ](https://github.com/appwrite/playground-for-go/pull/3/files#diff-fcc82ca569b20f8e21654d678d2eae96e86a953aeff6c87b1a0200aa2944bf05), where you can see the same space formatting there too.

**Review by loks0n — APPROVED** [review]

**Pull request merged**